### PR TITLE
Add check/throw new error before deleting active addresses via sore-api

### PR DESCRIPTION
**Pull request by drjamesj**
### 1. Why is this change necessary?
This PR addresses issue #2117 which identifies unexpected behaviour when deleting an active address via the store-api.

### 2. What does this change do, exactly?
Before deleting an address, in addition to checking if it is the default billing or shipping address (existing check), a further check is made to determine if it's the current active billing or shipping address based on the customer context. If so, it will throw an error.

### 3. Describe each step to reproduce the issue or behaviour.
1. Create a new billing or shipping address for a logged in custom via store-api
2. Set the address as the active address by calling `PATCH /context`
3. Next delete the address by calling `DELETE /store-api/account/address/${addressId}`
4. Log out of the user
5. Log back into the user to be greeted with below error

```
Argument 1 passed to Shopware\Core\Checkout\Customer\CustomerEntity::setActiveBillingAddress() must be an instance of Shopware\Core\Checkout\Customer\Aggregate\CustomerAddress\CustomerAddressEntity, null given, called in /app/platform/src/Core/System/SalesChannel/Context/SalesChannelContextFactory.php on line 478
```

### 4. Please link to the relevant issues (if any).
#2117 

### 5. Checklist
I am a first time contributor here, please let me know which (if any) of the following are required.

- [ ] I have written tests and verified that they fail without my change -- please let me know if this is required for this PR, and if so any instructions on how to do so would be appreciated
- [X] I have squashed any insignificant commits
- [ ] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes -- let me know if this is necessary
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Core/Checkout/Customer/SalesChannel/DeleteAddressRoute.php [hunk 1]
```diff
@@ -73,6 +74,11 @@ public function delete(string $addressId, SalesChannelContext $context, Customer
     {
         $this->validateAddress($addressId, $context, $customer);
 
+        if($addressId === $context->getCustomer()->getActiveBillingAddress()->getId()
```
Comment (shyim) [inline src/Core/Checkout/Customer/SalesChannel/DeleteAddressRoute.php]: Can you add a test case for this case? Thanks :smile:

**Review by shyim — CHANGES_REQUESTED** [review]

**Comment by shyim** [comment]
And a changelog file? :smile:

**Comment by drjamesj** [comment]
@shyim Test case and changelog added. Please let me know any feedback :)

**Comment by shyim** [comment]
Thanks for your Contribution! :blue_heart:

**Pull request closed without merging**